lib/api/statistics.js

```diff
@@ -0,0 +1,117 @@
+/* global fetch */
+/**
+ * @fileOverview Module for interacting with the statistcs.json endpoint.
+ *
+ */
+const envPrefixMap = {
+  qa: 'qa.api',
+  staging: 'stg.api',
+  production: 'api'
+}
+
+// Default configuration
+const API_VERSION = 5.4
+const MAX_REQUESTED_IDS = 100;
+
+/**
+ * Calls the statistics.json API with provided options. Returns a promise that
+ * is fulfilled with an array of Results from the API or rejected with the error
+ * message.
+ *
+ * Example:
+ *
+ * getStatistics({
+ *  productIds: ['product1', 'product2', 'product3'],
+ *  environment: 'qa',
+ *  key: 'clients_api_key',
+ *  type: 'Reviews',
+ *  filters: {
+ *   ContentLocale: 'en_US'
+ *  }
+ * }).then(results => {
+ *   // Do something with results
+ * }, errorMsg => {
+ *   // Do something with error.
+ * })
+ *
+ * https://developer.bazaarvoice.com/docs/read/conversations/statistics/display/5_4
+ *
+ * @param  {Object} options object that contains options
+ * @param  {Array}  options.productIds - array of product IDs
+ * @param  {Array}  options.environment - the data environment
+ * @param  {Array}  options.key - client's api key
+ * @param  {string}  options.type - Using "Reviews" returns statistics for all
+ *         content, including syndicated content (if enabled on your API key).
+ *         If you only want statistics for reviews you own that were written for
+ *         the products specified, use "NativeReviews" instead.
+ * @param  {Object}  [options.filters] - Filters object. Keyed by filter name.
+ * @return {Promise} a promise that will be resolved with the raw results
+ *                   from the statistics call.
+ */
+function getStatistics ({ productIds, environment, key, type, filters = {} }) {
+  if (!productIds) {
+    throw new TypeError('productIds must be an array')
+  }
+  if (!environment) {
+    throw new TypeError(
+      'environment must be \'qa\', \'staging\', or \'production\''
+    )
+  }
+  if (!key) {
+    throw new TypeError('key must be provided')
+  }
+  if (!type) {
+    throw new TypeError('type must be \'Reviews\' or \'NativeReviews\'')
+  }
+
+  const envPrefix = envPrefixMap[environment]
+
+  const uri = `https://${envPrefix}.bazaarvoice.com/data/statistics.json` +
+            `?apiversion=${API_VERSION}` +
+            `&passkey=${key}` +
+            `&stats=${type}` +
+            Object.keys(filters)
+              .map(filter => `&filter=${filter}:${filters[filter]}`)
+              .join()
+
+  // Clone the productIds so we can manipulate it.
+  productIds = productIds.slice()
```

Comment:
I'd never done this before - interesting.  While researching I also found that ES6's new spread (`...`) operator can be used to do the same thing.  Check out the "A more powerful array literal" section in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator.

I don't know that I prefer one over the other, just wanted to share.